jetty-server/src/main/java/org/eclipse/jetty/server/AbstractNCSARequestLog.java

```diff
@@ -220,6 +221,23 @@ public void log(Request request, Response response)
             LOG.warn(e);
         }
     }
+    
+    /**
+     * Extract the user authentication
+     * @param request The request to extract from
+     * @return The string to log for authenticated user.
+     */
+    protected String getAuthentication(Request request)
+    {
+        Authentication authentication = request.getAuthentication();
+        
+        if (authentication instanceof Authentication.User)
+            return ((Authentication.User)authentication).getUserIdentity().getUserPrincipal().getName();
```

Comment:
no real idea but are we sure to not have any NPE here?

Reply:
`(null instanceof Authentication.User)` correctly evaluates to false.   
Or are you concerned about the chain of UserIdentity->UserPrincipal->Name.  We had that chain before, so I think it is good

Reply:
yup the chain but you're certainly right :-)